### PR TITLE
Delete post

### PR DESCRIPTION
**Pull request by mohammedali101**
User can click on Icon on the top right of a post and delete it
When icon is clicked, an alert dialog will appear asking the user to confirm deletion
After clicking "Yes" the post will successfully delete
Downside is that it is not reactive and requires user to refresh page to see post removed
Refer to video below

https://user-images.githubusercontent.com/91173297/200092660-c21e2752-2346-464e-bfc7-e2e2291ab8ea.mov

### CODE REVIEW

**Comment by tanvirahmed725** [comment]
Can you make it so the component reloads upon deletion?

**Comment by abe-mused** [comment]
> Downside is that it is not reactive and requires user to refresh page to see post removed

Any change we add at this point has to be the way we want the app to "eventually" function, so this has to be reactive.

> Can you make it so the component reloads upon deletion?

To expand on this comment, we can make the page refresh after a successful delete/create, but i think a better way of making the page reactive (without a reload, so the app looks more 2022) is to just append/remove the post from the `_posts` list upon a successful create/delete request.

File: lib/pages/post_widgets/post_widget.dart [hunk 1]
```diff
@@ -8,14 +8,20 @@ import 'package:linear/util/apis.dart';
...
+      required this.onLike,
+      required this.token});
   final Post post;
   final String token;
   final bool liked;
   final VoidCallback onLike;
 
   likeUnlikePost() {
-    final Future<Map<String, dynamic>> successfulMessage = likePost(post.postId, token);
+    final Future<Map<String, dynamic>> successfulMessage =
```
Comment (abe-mused) [inline lib/pages/post_widgets/post_widget.dart:23]: Rename this to `responseMessage` and keep it in the same line

File: lib/util/apis.dart [hunk 2]
```diff
@@ -2,8 +2,10 @@ import 'dart:developer' as developer;
 import 'package:http/http.dart' as http;
 import 'dart:convert';
 
-Future<Map<String, dynamic>> postCommunity(String communityName, String token) async {
-  const url = 'https://qgzp9bo610.execute-api.us-east-1.amazonaws.com/prod/community';
+Future<Map<String, dynamic>> postCommunity(
+    String communityName, String token) async {
+  const url =
+      'https://qgzp9bo610.execute-api.us-east-1.amazonaws.com/prod/community';
```
Comment (abe-mused) [inline lib/util/apis.dart:8]: Can you keep these on the same line the way they were?

File: lib/pages/post_widgets/post_widget.dart [hunk 3]
```diff
@@ -67,24 +74,56 @@ class PostWidget extends StatelessWidget {
...
                     ),
+                    PopupMenuButton(itemBuilder: (context) {
+                      return [
+                        const PopupMenuItem<int>(
+                          value: 0,
+                          child: Text("Delete"),
+                        ),
+                      ];
+                    }, onSelected: (value) {
+                      if (value == 0) {
```
Comment (abe-mused) [inline lib/pages/post_widgets/post_widget.dart:102]: 1. can you format this properly, something like:
```
PopupMenuButton(
    itemBuilder: (context) {
            ...
        },
    onSelected: (value) {
            ...
        }
),
```

2. This now shows the "delete" option for everyone? even if the user is not the author of the post?

File: lib/util/apis.dart [hunk 4]
```diff
@@ -20,17 +22,24 @@ Future<Map<String, dynamic>> postCommunity(String communityName, String token) a
...
-        return {'status': false, 'message': 'An error occurred while creating the community, please try again.'};
+      } else if ('${response.body}' ==
+          '{"message":"The community already exists!"}') {
+        return {'status': false, 'message': 'The community already exists!'};
+      } else {
+        return {
+          'status': false,
+          'message':
+              'An error occurred while creating the community, please try again.'
+        };
```
Comment (abe-mused) [inline lib/util/apis.dart:33]: all of this should go back to the way it was formatted before

**Review by abe-mused — CHANGES_REQUESTED** [review]

File: lib/util/apis.dart (hunk 2 above)
Comment (mohammedali101) [reply lib/util/apis.dart]: I copied the entire api file from development, then added the api call for deletePost and saved it as a new commit. This should resolve the issues with the lines in the file.

File: lib/util/apis.dart (hunk 4 above)
Comment (mohammedali101) [reply lib/util/apis.dart]: resolved

**Comment by mohammedali101** [comment]
new PR created for delete post:
https://github.com/abe-mused/AccountabilityAppFrontEnd/pull/79

**Pull request closed without merging**